src/main/java/com/beerjournal/breweriana/user/ImageUserService.java

```diff
@@ -0,0 +1,71 @@
+package com.beerjournal.breweriana.user;
+
+import com.beerjournal.breweriana.file.persistence.FileRepository;
+import com.beerjournal.breweriana.user.persistence.User;
+import com.beerjournal.breweriana.user.persistence.UserRepository;
+import com.beerjournal.breweriana.utils.Converters;
+import com.beerjournal.breweriana.utils.ImageValidator;
+import com.beerjournal.breweriana.utils.SecurityUtils;
+import com.beerjournal.infrastructure.error.BeerJournalException;
+import com.beerjournal.infrastructure.error.ErrorInfo;
+import com.mongodb.gridfs.GridFSDBFile;
+import lombok.RequiredArgsConstructor;
+import org.bson.types.ObjectId;
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.IOException;
+
+import static com.beerjournal.infrastructure.error.ErrorInfo.USER_FORBIDDEN_MODIFICATION;
+
+@Service
+@RequiredArgsConstructor
+class ImageUserService {
+
+    private final UserRepository userRepository;
+    private final FileRepository fileRepository;
+    private final SecurityUtils securityUtils;
+    private final ImageValidator imageValidator;
+
+    void saveUserAvatarImage(MultipartFile multipartFile, String userId) throws IOException {
```

Comment:
I think this method should at least return id or even whole UserDto (with inserted image id) so that it could be added to the response body in controller and returned to user :)